File: ac044ccd-3d72-4455-8c1c-58b7a9637580-f03a43a3-5f6a-4811-aa2c-6c609cec2ffb-main/springapp/src/main/java/com/examly/springapp/service/WiFiSchemeRequestService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.exceptions.WiFiSchemeRequestException;
import com.examly.springapp.model.WiFiSchemeRequest;

public interface WiFiSchemeRequestService {
    WiFiSchemeRequest addWiFiSchemeRequest(WiFiSchemeRequest wiFiSchemeRequest) throws WiFiSchemeRequestException;

    WiFiSchemeRequest getWiFiSchemeRequestById(Long wifiSchemeRequestId) throws WiFiSchemeRequestException;

    List<WiFiSchemeRequest> getAllWiFiSchemeRequests();

    WiFiSchemeRequest updateWiFiSchemeRequest(Long wifiSchemeRequestId, WiFiSchemeRequest wiFiSchemeRequest);

    boolean deleteWiFiSchemeRequest(Long wifiSchemeRequestId);

    List<WiFiSchemeRequest> getWiFiSchemeRequestByUserId(Long userId);

}

File: ac044ccd-3d72-4455-8c1c-58b7a9637580-f03a43a3-5f6a-4811-aa2c-6c609cec2ffb-main/angularapp/src/app/services/wifi-scheme-request.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { WifiSchemeRequest } from '../models/wifi-scheme-request.model';

@Injectable({
  providedIn: 'root'
})

export class WifiSchemeRequestService {
  apiUrl: string = "https://8080-fadbadfaddcabbbeddcbfaafaaacccecffb.premiumproject.examly.io/api/wifiSchemeRequest";

  constructor(private http: HttpClient) { }

  getAllWiFiSchemeRequests():Observable<WifiSchemeRequest[]>{
    return this.http.get<WifiSchemeRequest[]>(this.apiUrl);
  }

  getWiFiSchemeRequestsByUserId(userId: number):Observable<WifiSchemeRequest[]>{
    return this.http.get<WifiSchemeRequest[]>(this.apiUrl+"/user/"+userId);
  }

  addWiFiSchemeRequest(wifiSchemeRequest: WifiSchemeRequest):Observable<WifiSchemeRequest>{
    return this.http.post<WifiSchemeRequest>(this.apiUrl, wifiSchemeRequest);
  }

  updateWiFiSchemeRequest(wifiSchemeRequestId: number, wifiSchemeRequest: WifiSchemeRequest):Observable<WifiSchemeRequest>{
    return this.http.put<WifiSchemeRequest>(this.apiUrl+"/"+wifiSchemeRequestId, wifiSchemeRequest);
  }

  deleteWiFiSchemeRequest(wifiSchemeRequestId: number):Observable<WifiSchemeRequest>{
    return this.http.delete<WifiSchemeRequest>(this.apiUrl+"/"+wifiSchemeRequestId);
  }

}
